crates/fuel-core/src/graphql_api.rs

```diff
@@ -168,3 +169,15 @@ impl<T> IntoApiResult<T> for Result<T, StorageError> {
         }
     }
 }
+
+pub fn require_historical_execution(ctx: &Context<'_>) -> async_graphql::Result<()> {
+    let config = ctx.data_unchecked::<Config>();
+
+    if config.historical_execution {
+        Ok(())
+    } else {
+        Err(async_graphql::Error::new(
+            "`--historical-execution` is required for this operation",
```

Comment:
nit: From what I understand this error is returned to the user. I don't think an API user will be informed about the flags of the CLI and it also cause a potential tech-debt if we change flag name. I think we should provide a more generalized sentence to the user such as : `Historical execution isn't activated on this API` smth like that

Reply:
We do the same for `debug`, but I think the suggesting here makes sense. Are you ok if we leave that as a follow-up, and change refrences to debug as well?

Reply:
Yep